### DISCUSSION TITLE
Label too big compared to the text inside

### DISCUSSION THREAD

**Discussion by FlorianPasDidee** [TomSchimansky/CustomTkinter#2847] [Q&A]
Hello everyone, 

When I use a CTkLabel for a text, the label appears to be much larger than the text.
How can I reduce the size of the label as close as possible to the size of the text ? Is this a bug ? 
I searched the net and tried to force the size of the label in several ways but it doesn't work... 

All tracks are welcome ! THANKS

Here is an example of what I said:

The part of code :
![Capture d'écran 2024-08-01 142407](https://github.com/user-attachments/assets/88470c10-8ec0-47eb-8834-2cd26cb2d0d5)

What it produce... :
![Capture d'écran 2024-07-31 221543](https://github.com/user-attachments/assets/1d1978c6-b032-40cb-a66d-209294842580)

The text is the right size, it is the label (here the blue rectangle) which is too high.

**Comment by ThomasRadigk** [comment]
Hi,
don't use `sticky="wns"` - this means it is stretched from north to south - it will occupy the whole vertical space, which you don't want. You only want `sticky='w'`. Then the height actually will define the size of the blue background in your case. But the label object does not know about the size of the text, so you would have to adjust it with the height property yourself.
note: there also is an anchor property for the label, which defaults to 'center'.

If you had copied your code, and not posted screenshots, I would have given you the adjusted code.

**Comment by FlorianPasDidee** [comment]
Hi @ThomasRadigk,

I have already tested this but it doesn't change anything, whether I put "w", "wn", "ws" or "wns" the result is always the same. 
The blue background is always larger than the text, even if I force its size with the "height" parameter. 
Likewise with the "anchor" parameter. I tested with the "Arial" font and the spaces on either side of the text seem less important for the same font size (see texts with red background).

I have been using customtkinter for several years and I have never encountered this type of problem which seems unsolvable...

Thanks for your help

I updated my code:


    self.text_equipe_main = CTkLabel(app.boite_affichage, height=200, width=780, text=self.nom_equipe_main,
                                            font=("Bowlby One SC", 190), text_color="#f2e5cc", bg_color="blue", anchor="n")
    self.text_equipe_main.grid(row=1, column=1, padx=35, sticky="w")



The test with different fonts:



    *With Bowlby One SC:*
    self.text_classement = CTkLabel(self.boite_infos, text=self.classement_equipe_main, font=("Bowlby One SC", 140),
                                        text_color="#b6ad9b", bg_color="red")
     self.text_classement.grid(row=0, column=1, padx=25, sticky="nw")
    

    *With Arial*
    self.text_classement = CTkLabel(self.boite_infos, text=self.pts_equipe_main, font=("arial", 140),
                                        text_color="#b6ad9b", bg_color="red")
    self.text_classement.grid(row=0, column=3, padx=20, sticky="nw")

What it produce:
![Capture d'écran 2024-08-07 141419](https://github.com/user-attachments/assets/12582e1e-67d6-41aa-85d1-ad0ac36c34fd)

**Comment by FedericoSpada** [comment]
Hi,
I think that the font itself has some white space on top and bottom of all letters. Or maybe other widgets on the same row have a bigger height, and the label is stretched to the same height.
Maybe you can get around the problem by creating an image with the text you want and display that one.
